nextcord/ui/select/channel.py

```diff
@@ -0,0 +1,257 @@
+from ...enums import ComponentType
+from ...utils import MISSING, get
+from ..item import Item, ItemCallbackType
+from .string import Select
+
+if TYPE_CHECKING:
+    from ...abc import GuildChannel
+    from ...enums import ChannelType
+    from ...guild import Guild
+    from ...threads import Thread
+    from ...types.components import ChannelSelectMenu as ChannelSelectMenuPayload
+
+__all__ = ("ChannelSelect", "channel_select")
+
+
+class ChannelSelectValues(UserList):
+    """Represents the values of a :class:`ChannelSelect`."""
+
+    def get(self, guild: Guild) -> List[Union[GuildChannel, Thread]]:
+        """A shortcut for getting all :class:`nextcord.abc.GuildChannel`'s of :attr:`.values`.
+
+        Channels that are not found in cache will not be returned.
+        To get all channels regardless of whether they are in cache or not, use :meth:`.fetch_channels`.
+
+        Parameters
+        ----------
+        guild: :class:`nextcord.Guild`
+            The guild to get the channels from.
+
+        Returns
+        -------
+        List[Union[:class:`nextcord.abc.GuildChannel`, :class:`nextcord.Thread`]]
+            A list of channels that have been selected by the user.
+        """
+        channels: List[Union[GuildChannel, Thread]] = []
+        for id in self.data:
+            channel = guild.get_channel_or_thread(id)
+            if channel is not None:
+                channels.append(channel)
+        return channels
+
+    async def fetch(self, guild: Guild) -> List[Union[GuildChannel, Thread]]:
+        """A shortcut for fetching all :class:`nextcord.abc.GuildChannel`'s of :attr:`.values`.
+
+        Channels that are not found in cache will be fetched.
+
+        Parameters
+        ----------
+        guild: :class:`nextcord.Guild`
+            The guild to get the channels from.
+
+        Raises
+        ------
+        :exc:`.HTTPException`
+            Fetching the channels failed.
+        :exc:`.NotFound`
+            A channel was not found.
+        :exc:`.Forbidden`
+            You do not have the proper permissions to fetch a channel.
+        :exc:`.InvalidArgument`
+            The channel type is not supported.
+
+        Returns
+        -------
+        List[Union[:class:`nextcord.abc.GuildChannel`, :class:`nextcord.Thread`]]
+            A list of channels that have been selected by the user.
+        """
+        channels: List[Union[GuildChannel, Thread]] = []
+        guild_channels = None
+        for id in self.data:
+            channel = guild.get_channel_or_thread(id)
+            if channel is None:
+                if guild_channels is None:
+                    guild_channels = await guild.fetch_channels()
+                channel = get(guild_channels, id=id)
+                if channel:
+                    channels.append(channel)
+            else:
+                channels.append(channel)
+        return channels
+
+
+class ChannelSelect(Select):
```

Comment:
These classes ChannelSelect, UserSelect, etc. should probably appear in api.rst with the other UI kit classes